iceoryx2-cli/iox2/tests/commands_tests.rs

```diff
@@ -0,0 +1,153 @@
+        fs::{metadata, set_permissions},
+        os::unix::fs::PermissionsExt,
+    };
+
+    #[cfg(windows)]
+    use std::io::Write;
+
+    #[cfg(windows)]
+    const MINIMAL_EXE: &[u8] = include_bytes!("minimal.exe");
+
+    const IOX2_PREFIX: &str = "iox2-";
+    const FOO_COMMAND: &str = "Xt7bK9pL";
+    const BAR_COMMAND: &str = "m3Qf8RzN";
+    const BAZ_COMMAND: &str = "P5hJ2wAc";
+
+    macro_rules! create_file {
+        ($path:expr, $file:expr) => {{
+            let file_path = $path.join($file);
+            File::create(&file_path).expect("Failed to create file");
+            #[cfg(unix)]
+            {
+                let mut perms = metadata(&file_path)
+                    .expect("Failed to get metadata")
+                    .permissions();
+                perms.set_mode(0o755);
+                set_permissions(&file_path, perms).expect("Failed to set permissions");
+            }
+            #[cfg(windows)]
+            {
+                const COMMAND_EXT: &str = "exe";
+
+                let mut file = File::create(&file_path).expect("Failed to create file");
+
+                if file_path
+                    .extension()
+                    .and_then(|ext| ext.to_str())
+                    .unwrap_or("")
+                    == COMMAND_EXT
+                {
+                    file.write_all(&MINIMAL_EXE)
+                        .expect("Failed to write executable content");
+                }
+            }
+        }};
+    }
+
+    struct TestEnv {
+        _temp_dir: TempDir,
+        original_path: String,
+    }
+
+    impl TestEnv {
+        fn setup() -> Self {
+            let original_path = env::var("PATH").expect("Failed to get PATH");
+
+            let temp_dir = TempDir::new().expect("Failed to create temp dir");
+            let temp_path = temp_dir.path().to_path_buf();
+
+            let mut paths = env::split_paths(&original_path).collect::<Vec<_>>();
+            paths.push(temp_path.clone());
+            let new_path = env::join_paths(paths).expect("Failed to join paths");
+            env::set_var("PATH", &new_path);
+
+            create_file!(temp_path, format!("{}{}", IOX2_PREFIX, FOO_COMMAND));
+            create_file!(temp_path, format!("{}{}.d", IOX2_PREFIX, FOO_COMMAND));
+            create_file!(temp_path, format!("{}{}.exe", IOX2_PREFIX, FOO_COMMAND));
+            create_file!(temp_path, format!("{}{}", IOX2_PREFIX, BAR_COMMAND));
+            create_file!(temp_path, format!("{}{}.d", IOX2_PREFIX, BAR_COMMAND));
+            create_file!(temp_path, format!("{}{}.exe", IOX2_PREFIX, BAR_COMMAND));
+            create_file!(temp_path, BAZ_COMMAND);
+            create_file!(temp_path, format!("{}.d", BAZ_COMMAND));
+            create_file!(temp_path, format!("{}.exe", BAZ_COMMAND));
+
+            TestEnv {
+                _temp_dir: temp_dir,
+                original_path,
+            }
+        }
+    }
+
+    impl Drop for TestEnv {
+        fn drop(&mut self) {
```

Comment:
Maybe I am missing it, but do you remove the files after the test run?

Reply:
@elfenpiff The entire directory is dropped when the `TestEnv` goes out of scope, handled automatically by `tempfile`.